docs/FeatureManagement/Clients/Microsoft.FeatureManagement.AspNetCore.md

```diff
@@ -0,0 +1,367 @@
+
+This builds off the more generic capability to branch the entire application based on a feature.
+
+```
+app.UseForFeature(featureName, appBuilder => 
+{
+    appBuilder.UseMiddleware<T>();
+});
+```
+
+### Disabled Action Handling
+
+When an MVC action requires to be enabled and the feature is not enabled, a registered `IDisabledFeatureHandler` will be invoked. By default, a minimalistic handler is registered which returns HTTP 404. This can be overridden using the `IFeatureManagmentBuilder` when registering feature flags.
+
+## Implementing a Feature Filter
+
+Creating a feature filter provides a way to enable features based on criteria that you define. To implement a feature filter, the `IFeatureFilter` interface must be implemented. `IFeatureFilter` has a single method named `Evaluate`. When a feature specifies that it can be enabled for a feature filter, the `Evaluate` method is called. If `Evaluate` returns `true` it means the feature should be enabled.
+
+Feature filters are registered by the `IFeatureManagementBuilder` when `AddFeatureManagement` is called. These feature filters have access to the services that exist within the service collection that was used to add feature flags. Dependency injection can be used to retrieve these services.
+
+### Parameterized Feature Filters
+
+Some feature filters require parameters to decide whether a feature should be turned on or not. For example a browser feature filter may turn on a feature for a certain set of browsers. It may be desired that Edge and Chrome browsers enable a feature, while FireFox does not. To do this a feature filter can be designed to expect parameters. These parameters would be specified in the feature configuration, and in code would be accessible via the `FeatureFilterEvaluationContext` parameter of `IFeatureFilter.Evaluate`.
+
+```
+  public class FeatureFilterEvaluationContext
+  {
+      /// <summary>
+      /// The name of the feature being evaluated.
+      /// </summary>
+      public string FeatureName { get; set; }
+
+      /// <summary>
+      /// The settings provided for the feature filter to use when evaluating whether the feature should be enabled.
+      /// </summary>
+      public IConfiguration Parameters { get; set; }
+  }
+```
+
+`FeatureFilterEvaluationContext` has a property named `Parameters`. These parameters represent a raw configuration that the feature filter can use to decide how to evaluate whether the feature should be enabled or not. To use the browser feature filter as an example once again, the filter could use `Parameters` to extract a set of allowed browsers that would have been specified for the feature and then check if the request is being sent from one of those browsers.
+
+```
+  [FilterAlias("Browser")]
```

Comment:
The "Browser" will just work even without this FilterAlias, right? Should we explain the FilterAlias attribute a little bit here?